### PR TITLE
Add support for calling API Key endpoints.

### PR DESCRIPTION
**Pull request by orendain**
Part of https://github.com/grafana/terraform-provider-grafana/issues/143.

As per the issue above, this PR exposes methods for Creating/Retrieving/Deleting API keys by calling Grafana's Authorization API.

A fork of this client, with the changes implemented, is already being used by https://github.com/orendain/terraform-provider-grafanaauth.  Before that functionality can be contributed back to the official Grafana Terraform Provider, this client needs these updates.

I tried to keep the implementation consistent with the rest of the project, and happy to make changes that may be required.

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -107,7 +107,7 @@ func (c *Client) newRequest(method, requestPath string, query url.Values, body i
 	if c.config.APIKey != "" {
 		req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", c.config.APIKey))
 	} else if c.config.OrgID != 0 {
-		req.Header.Add("X-Grafana-Org-Id", strconv.FormatInt(c.config.OrgID, 10))
+		req.Header.Add("X-Grafana-Org-ID", strconv.FormatInt(c.config.OrgID, 10))
```
Comment (aknuds1) [inline client.go:110]: Why this change? It is written `X-Grafana-Org-Id` in the Grafana source code.

File: api_key_test.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		{
+			"id": 2,
+			"name": "key-name-2",
+			"role": "Admin",
+			"expiration": "2021-10-30T10:52:03+03:00"
+		}
+	]`
+)
+
+func TestCreateApiKey(t *testing.T) {
```
Comment (aknuds1) [inline api_key_test.go:28]:
```suggestion
func TestCreateAPIKey(t *testing.T) {
```

File: api_key_test.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+	res, err := client.CreateAPIKey(req)
+	if err != nil {
+		t.Error(err)
+	}
+
+	t.Log(pretty.PrettyFormat(res))
+}
+
+func TestDeleteApiKey(t *testing.T) {
```
Comment (aknuds1) [inline api_key_test.go:46]:
```suggestion
func TestDeleteAPIKey(t *testing.T) {
```

File: api_key_test.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+	res, err := client.DeleteAPIKey(int64(1))
+	if err != nil {
+		t.Error(err)
+	}
+
+	t.Log(pretty.PrettyFormat(res))
+}
+
+func TestGetApiKeys(t *testing.T) {
```
Comment (aknuds1) [inline api_key_test.go:58]:
```suggestion
func TestGetAPIKeys(t *testing.T) {
```

**Comment by julienduchesne** [comment]
https://github.com/grafana/grafana-api-golang-client/pull/39 supersedes this one
